cdci_data_analysis/analysis/parameters.py

```diff
@@ -218,12 +230,22 @@ def default_units(self):
         return self._default_units
 
     @default_units.setter
-    def default_units(self, units):
+    def default_units(self, par_unit):
+        if self._allowed_types is not None:
+            self.chekc_units(par_unit, self._allowed_units, self.name)
```

Comment:
Oh, maybe since we are deep into this, can we renamed `chekc_units`? With compatibility, by adding this:

```
def chekc_units(self, *args, **kwargs):
    warning('please update to new interface! -- ....')
    return self.check_units(*args, **kwargs)
```

and only using `check_units` in new things.